spec/src/main/asciidoc/microprofile-lra-spec.adoc

```diff
@@ -801,6 +807,11 @@ Users are allowed to reuse existing JAX-RS endpoints for participant method defi
 In this case the LRA implementation MUST ensure that invoking these methods outside of the 
 implementation of the LRA specification will not influence any running LRA.
 
+Specifically, developers should NEVER call any JAX-RS endpoint for participant methods where they
+added a `Context Header` (specified by the Java constant `LRA_HTTP_CONTEXT_HEADER`)
```

Comment:
Hmm, `@AfterLRA` could be a problem though.

Reply:
with 
>participant methods

I meant all methods on the participant level which are called by the implementation (complete, compensate, status, afterLRA). I'll update the text to make it more clear. (although status is not a real issue but best to keep it general).

Reply:
>This should be LRA_HTTP_RECOVERY_HEADER

The compensate and complete methods expect LRA.LRA_HTTP_CONTEXT_HEADER constant in the header, so this is correct.

But need to adjust for AfterLRA where the id is sent by means of another header.

So I'll make it clear that developer should NEVER call a participant method (but of course, is allowed to set the header when calling 'regular' endpoints.

Reply:
> I meant all methods on the participant level which are called by the implementation (complete, compensate, status, afterLRA). I'll update the text to make it more clear. (although status is not a real issue but best to keep it general).

For participant callback methods (Compensate, Complete, Status and Forget) the participant should check for the presence of LRA_HTTP_RECOVERY_HEADER. If that header is not present it should treat the invocation as a regular business method. If it does contain the header it should process the request as per our specification.

Reply:
I'll update the wording around the LRA_HTTP_RECOVERY_HEADER then because now it is not mandatory to generate this header for recovery means (Because some implementations do not need this - as discussed several months ago - in order to be able to recover after application restart on different URL)

I'll create a separate PR for this SPEC update and the updates to the TCK to test out this requirement.